Code/GraphMol/FileParsers/MolFileWriter.cpp

```diff
@@ -800,27 +800,60 @@ void GetMolFileBondStereoInfo(const Bond *bond, const INT_MAP_INT &wedgeBonds,
             ROMol::OEDGE_ITER beg, end;
             boost::tie(beg, end) =
                 bond->getOwningMol().getAtomBonds(bond->getBeginAtom());
+            bool atomHasDuplicateRanks = false;
+            std::vector<int> nbrRanks;
             while (beg != end && !nbrHasDir) {
-              const Bond *nbrBond = bond->getOwningMol()[*beg];
-              if (nbrBond->getBondType() == Bond::SINGLE &&
-                  (nbrBond->getBondDir() == Bond::ENDUPRIGHT ||
-                   nbrBond->getBondDir() == Bond::ENDDOWNRIGHT)) {
-                nbrHasDir = true;
+              const auto nbrBond = bond->getOwningMol()[*beg];
```

Comment:
We are doing the same for the begin and the end atom, right? For the sake of maintainability, I think this has reached the point where it is worth abstracting into a separate function and calling it on each end.